src/mm/taskmem.rs

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+//
+// Copyright (c) 2022-2023 SUSE LLC
+//
+// Author: Roy Hopkins <rhopkins@suse.de>
+
+use crate::{
+    address::VirtAddr, cpu::percpu::this_cpu, error::SvsmError, task::TaskError, types::PAGE_SIZE,
+};
+
+/**
+ * Allocate a virtual address range. The range is not mapped
+ * to any physical address.
+ */
+pub fn valloc(size: usize) -> Result<VirtAddr, SvsmError> {
+    // Align size up to next page boundary
+    let size = (size + PAGE_SIZE - 1) & !(PAGE_SIZE - 1);
+    let t = this_cpu()
```

Comment:
The whole `this_cpu() ... current_task()` happens often enough to make me think we should have a helper function `current()`.